### DISCUSSION TITLE
[github.dev] Communicate clearly that working copy in the browser is short-lived

### DISCUSSION THREAD

**Discussion by LinqLover** [community/community#12797] [Codespaces] [answered]
I was pretty irritated that after returning from my break, my not-so-small working copy of a larger documentation in a github.dev tab was lost because my PC had installed updates and restarted. If a web service does NOT have server-side storage (to my understanding, server-side storage is the implicit norm today), IMHO this should be communicated clearly, e.g., via a small pop-up at the top of the screen (idc you could even place an advertisement to commercial Codespaces there). Unexpected data loss is just so 20th century. Thanks for your time!

**Comment by GitTom** [comment]
I guess the server-side storage would be a repo.

But I think your current working state - stuff that hasn't been pushed to a repo - is stored in the browser's local storage, so you shouldn't have lost it.

**Reply by LinqLover** [reply, marked as answer]
The issue is 1y old, I haven't experienced the same problem again since then. Quite the opposite, now github.dev even warns when I want to close the browser tab *after* having committed and pushed all changes. :-)

**Reply by joshspicer** [reply]
I just was typing a response now that this feedback was surfaced again - glad to hear the product has improved for you :)
